promql/functions.go

```diff
@@ -1003,40 +1053,55 @@ func funcHistogramQuantile(vals []parser.Value, args parser.Expressions, enh *Ev
 
 // === resets(Matrix parser.ValueTypeMatrix) Vector ===
 func funcResets(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
```

Comment:
I don't think the implementation is right. We are not considering the case where a series switched from float, to histogram, to float, again to histogram, for example. And when there is a mix of type like this, I don't think there can be a right answer. Since this is unrelated to this refactor (earlier we considered only float samples; this changes the function behaviour), I suggest this to be taken into another PR.

Reply:
Yeah, I, too, believe that there is no right answer for mixed series. The currently implemented behavior is documented (in the part you haven't reviewed yet), and maybe that's the best we can do now.

(It's one of the reasons why I think we should avoid mixed series in the first place, cf. our brainstorming in Monaco, which we should tackle once we got all the other basic histogram TODOs done.)